fortex/health/processors/scispacy_processor.py

```diff
@@ -0,0 +1,170 @@
+                        f"{abrv} \t ({abrv.start}, {abrv.end}) {abrv._.long_form}"
+                    )
+                    tmpAbrv = Abbreviation(
+                        pack=input_pack, begin=abrv.start, end=abrv.end
+                    )
+                    tmpAbrv.long_form = abrv._.long_form
+                    list_of_abrvs.append(tmpAbrv)
+
+            else:
+                for item in doc._.hearst_patterns:
+                    general_concept: Phrase = Phrase(
+                        pack=input_pack, begin=item[1].start, end=item[1].end
+                    )
+                    specific_concept = Phrase(
+                        pack=input_pack, begin=item[2].start, end=item[2].end
+                    )
+                    hlink = Hyponym(
+                        pack=input_pack,
+                        parent=general_concept,
+                        child=specific_concept,
+                    )
+                    hlink.hyponym_link = item[0]
+
+    @classmethod
+    def default_configs(cls):
+        r"""
+        This defines a basic config structure for `ICDCodingProcessor`.
+
+        Following are the keys for this dictionary:
+         - `entry_type`: input entry type,
+         - `model_name`: the higgingface transformer model name to be
+                         used for classification,
+
+        Returns: A dictionary with the default config for this processor.
+        """
+        return {
+            "entry_type": "ft.onto.base_ontology.Document",
+            "attribute_name": "classification",
+            "multi_class": True,
+            "model_name": "en_core_sci_sm",
+            "pipe_name": "abbreviation_detector",
+            "cuda_devices": -1,
+        }
+
+    def expected_types_and_attributes(self):
+        r"""
+        Method to add user specified expected type which would be checked
+        before running the processor if the pipeline is initialized with
+        `enforce_consistency=True` or
+        :meth:`~forte.pipeline.Pipeline.enforce_consistency` was enabled for
+        the pipeline.
+        """
+        return {
+            "ft.onto.base_ontology.Document": set(),
+            "forte.data.ontology.top.Annotation": set(),
+        }
+
+    def record(self, record_meta: Dict[str, Set[str]]):
+        r"""
+        Method to add output type record of `ICDCodeProcessor` which
+        is `"ftx.medical.clinical_ontology.MedicalArticle"` with attributes:
+         `icd_version` and `icd_code`
+        to :attr:`forte.data.data_pack.Meta.record`.
+
+        Args:
+            record_meta: the field in the datapack for type record that need to
+                fill in for consistency checking.
+        """
+        record_meta["ftx.medical.clinical_ontology.Abbreviation"] = {
+            "long_form",
+        }
+        record_meta["ftx.medical.clinical_ontology.hyponym"] = {
+            "hyponym_link",
+            "ParentType",
+            "ChildType",
```

Comment:
I don't think the class variables (ParentType, ChildType) need to be added here.

Reply:
The ParentType and ChildType is now removed.